### PR TITLE
Remove checking duckdb_free

### PR DESCRIPTION
**Pull request by Ashvith**
This should solve issue #301. Not sure if I have to remove `#ifdef HAVE_DUCKDB_FREE`  from `result.c`.

### CODE REVIEW

**Comment by suketa** [comment]
@Ashvith 
> Not sure if I have to remove #ifdef HAVE_DUCKDB_FREE from result.c.

Remove #ifdef HAVE_DUCKDB_FREE and #elsif block.
removing have_func invokes `free` instead of `duckdb_free` and occurs segmentation error. (C-level error.)

**Comment by Ashvith** [comment]
Apologies for the wrong rename - it looks like the name for commit `c016451b985154c2fd6d33093be026a72e28d2ba`  should have been **fix** instead of **chore** according to the commit convention.

File: ext/duckdb/result.c [hunk 1]
```diff
@@ -65,11 +65,7 @@ static VALUE to_ruby_obj_string_from_blob(duckdb_result *result, idx_t col_idx,
     str = rb_str_new(bval.data, bval.size);
 
     if (bval.data) {
-#ifdef HAVE_DUCKDB_FREE
-        duckdb_free(bval.data);
```
Comment (suketa) [inline ext/duckdb/result.c:69]: Don't remove this line.

File: ext/duckdb/result.c [hunk 2]
```diff
@@ -65,11 +65,7 @@ static VALUE to_ruby_obj_string_from_blob(duckdb_result *result, idx_t col_idx,
...
-#else
         free(bval.data);
```
Comment (suketa) [inline ext/duckdb/result.c:68]: remove this line. We need `duckdb_free` but not `#else` *block*

File: ext/duckdb/result.c [hunk 3]
```diff
@@ -100,11 +96,7 @@ static VALUE to_ruby_obj(duckdb_result *result, idx_t col_idx, idx_t row_idx) {
         p = duckdb_value_varchar(result, col_idx, row_idx);
         if (p) {
             obj = rb_str_new2(p);
-#ifdef HAVE_DUCKDB_FREE
-            duckdb_free(p);
```
Comment (suketa) [inline ext/duckdb/result.c:104]: don't remove this line

File: ext/duckdb/result.c [hunk 4]
```diff
@@ -100,11 +96,7 @@ static VALUE to_ruby_obj(duckdb_result *result, idx_t col_idx, idx_t row_idx) {
...
-#else
             free(p);
```
Comment (suketa) [inline ext/duckdb/result.c:99]: remove this line. we need `duckdb_free` but not `free`

**Review by suketa — CHANGES_REQUESTED** [review]
@Ashvith 

Pls change the code

File: ext/duckdb/result.c (hunk 1 above)
Comment (Ashvith) [reply ext/duckdb/result.c:69]: Replaced free() with duckdb_free() in line 69.

File: ext/duckdb/result.c (hunk 3 above)
Comment (Ashvith) [reply ext/duckdb/result.c:104]: Replaced free() with duckdb_free() in line 104.

**Review by suketa — APPROVED** [review]
Thank you

**Pull request merged**